org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -168,10 +199,27 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 				} catch (CoreException e) {
 					JavaLanguageServerPlugin.logException("Build failed ", e);
 					connection.sendStatus(ServiceStatus.Error, getMessage(e.getStatus()));
+				} finally {
```

Comment:
we should also delay registration of codelens, signaturehelp  support

Reply:
They are alredy delayed. See https://github.com/snjeza/eclipse.jdt.ls/blob/d3565e35a3ca52fff272fb9341f32bd995f23449/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java#L224

Reply:
ok fair enough, so, to be consistent -and accurate-, all the following statements should also be put under JDTLanguageServer.didChangeConfiguration